### PR TITLE
Use Docker

### PR DESCRIPTION
**Pull request by mgaeta**
Instead of using a JavaScript action, build with a Dockerfile and release as a Docker action.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+COPY --from=builder /app/package.json /app/yarn.lock /action-release/
+RUN export YARN_CACHE_FOLDER="$(mktemp -d)" \
+  && cd /action-release \
+  && yarn install --frozen-lockfile --production --quiet \
+  && rm -r "$YARN_CACHE_FOLDER"
+
+
+COPY --from=builder /app/dist /action-release/dist/
+RUN chmod +x /action-release/dist/index.js
+#RUN ln -s /action-release/dist/index.js ./dist/index.js
```
Comment (manuzope) [inline Dockerfile:20]: Do we need this comment?

File: .gitignore [hunk 2]
```diff
@@ -78,6 +78,9 @@ typings/
 # nuxt.js build output
 .nuxt
 
+# Vercel build output
```
Comment (manuzope) [inline .gitignore:81]: Is this Vercel? Or just yarn?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+FROM node:12-buster as builder
+WORKDIR /app
+COPY package.json yarn.lock ./
+RUN export YARN_CACHE_FOLDER="$(mktemp -d)" \
+  && yarn install --frozen-lockfile --quiet \
+  && rm -r "$YARN_CACHE_FOLDER"
+COPY . .
+RUN yarn build
+
+FROM node:12-buster
```
Comment (NisanthanNanthakumar) [inline Dockerfile:10]: @mgaeta I think if you use the alpine version, it will make the overall image smaller.

File: .gitignore (hunk 2 above)
Comment (mgaeta) [reply .gitignore:81]: `yarn build` runs `ncc build` so I think it's a Vercel thing.

File: Dockerfile (hunk 3 above)
Comment (mgaeta) [reply Dockerfile]: It looks like alpine worked in the test workflow!

**Pull request merged**